ui/src/app/views/workspace-wrapper/component.ts

```diff
@@ -69,7 +66,18 @@ export class WorkspaceWrapperComponent implements OnInit, OnDestroy {
         // do not render the child component with a stale workspace.
         this.workspace = undefined;
         this.accessLevel = undefined;
-        return this.workspaceStorageService.getWorkspace(ns, wsid);
+        // This needs to happen for testing. Otherwise it tries to make an api
+        // call with undefined, because the component initializes before we have
+        // access to the route.
+        if (ns === undefined || wsid === undefined) {
+          return Promise.resolve({name: '', accessLevel: WorkspaceAccessLevel.NOACCESS});
```

Comment:
Returning a stub workspace seems problematic here. I might recommend just returning `Observable.empty();` which would not trigger the next callback.

Reply:
Agreed that it could be problematic. However `Observable.empty()` did actually trigger the next callback. I am going to try adding an empty object check in the subscribe, and returning an empty object.

Reply:
OK, that didn't work then. I'd just use null instead. I think passing `Observable.empty()` as a sentinel value would be pretty confusing.